getDataTableReader.html

```diff
@@ -0,0 +1,159 @@
+
+        // Loop through each table that was used in creating this data source
+        logicalTables.forEach(function (table) {
+          console.log(`Name of logical table: ${table.caption}`);
+          document.getElementById(
+            'info'
+          ).innerHTML = `<p>Name of logical table: ${table.caption}</p>`
+        });
+
+        // Get the data from the first logical table
+        const logicalTableData = await datasource[0].getLogicalTableDataAsync(
+          logicalTables[0].id
+        );
+        console.log(logicalTableData);
+        console.log(`Total row count: ${logicalTableData.totalRowCount}`);
+        // Display the row count; the maximum is 10,000 rows
+        document
+          .getElementById('info')
+          .insertAdjacentHTML(
+            'beforeend',
+            `<p>Total row count: ${logicalTableData.totalRowCount}</p>`
+          );
+
+        // Set the page size to 1000 for this sample; the default value is 10000 rows
+        const pageRowCount = 1000;
+        // Create the data table reader
+        const pageReader = await datasource[0].getLogicalTableDataReaderAsync(
+          logicalTables[0].id,
+          pageRowCount
+        );
+        try {
+        // Display the number of pages to read
+        console.log(`Number of pages: ${pageReader.pageCount}`);
+        document
+          .getElementById('info')
+          .insertAdjacentHTML(
+            'beforeend',
+            `<p>Number of pages: ${pageReader.pageCount}</p>`
+          );
+
+        // Get the page data based on page number requested
+        let currentPageDataTable = await pageReader.getPageAsync(pageNumber);
+        console.log(currentPageDataTable);
+
+        // Add data to HTML element.
+        const target = document.getElementById('dataTarget')
+        target.innerHTML = `<h4>Data Table Page ${pageNumber} Data:</h4><p>${JSON.stringify(
+          currentPageDataTable.data
+        )}</p>`;
+
+        } catch (e) {
+          console.error(e);
+
+        } finally {
+        // Free up memory when done
+        await pageReader.releaseAsync();
+        }
+      }
+
+      // get the viz from the HTML web component
+      const viz = document.getElementById('tableauViz');
+
+      // Event fired when a viz first becomes interactive.
+      viz.addEventListener(
+        TableauEventType.FirstInteractive,
+        handleFirstInteractive
+      );
+    </script>
+  </head>
+
+  <body>
+    <div>
+      <h1>Get Logical Data (TableDataReader) Example </h1>
+      <p>
+        Click <b>Get Data</b> to get the first page (page 0) of the logical
+        table data for the viz.
+      </p>
+      <button id="get-data">Get Data</button>
+      <div id="info"></div>
+      <p>Select a page to retrieve additional data:</p>
+      <select
+        id="pageSelect"
+        class="form-select"
+        aria-label="Default select example"
+        disabled="false"
```

Comment:
Counterintuitively, `disabled="false"` actually disables the dropdown. `disabled` is a boolean attribute so it even existing disables the element. The value is ignored.

Reply:
Oh. That makes sense now. Thanks.